### PR TITLE
remote: add basic cl_pocl_content_size support

### PR DESCRIPTION
**Pull request by jansol**
This was left out from the initial remote driver PR.

+ a couple of small tweaks

### CODE REVIEW

**Review by pjaaskel — CHANGES_REQUESTED** [review]
Otherwise looks good, but can you add a mention in the Remote part of the user manual since this is an essential optimization feature to use with PoCL-R and compressed data?

File: lib/CL/pocl_util.c [hunk 1]
```diff
@@ -675,12 +675,19 @@ pocl_create_command_struct (_cl_command_node **cmd,
   return errcode;
 }
 
+/**
+ * @param dev Destination device
+ * @param ev_export_p Optional output parameter for the export event
+ * @param migration_size Max number of bytes to migrate (caller has to read
+ *                       content size from mem->size_buffer if applicable)
```
Comment (pjaaskel) [inline lib/CL/pocl_util.c:682]: Set to 0 if not applicable?

File: tests/runtime/test_cl_pocl_content_size_migration.c [hunk 2]
```diff
@@ -0,0 +1,153 @@
+/* Test "cl_pocl_content_size" PoCL extension
+
+   Copyright (C) 2021 Tampere University
```
Comment (pjaaskel) [inline tests/runtime/test_cl_pocl_content_size_migration.c:3]:
```suggestion
   Copyright (C) 2021-2023 Tampere University
```

**Review by pjaaskel — APPROVED** [review]

File: lib/CL/pocl_util.c (hunk 1 above)
Comment (jansol) [reply lib/CL/pocl_util.c:682]: Currently the caller just sets it to mem->size in that case, and a migration_size of 0 causes the migration to behave as if CL_MIGRATE_MEM_OBJECT_CONTENT_UNDEFINED was given, which seemed like the most sensible thing to do. It is after all entirely possible that a kernel produces no output e.g. due to incomplete input data or due to this being an auxiliary buffer that happens to be unneeded for a given invocation.

**Comment by franz** [comment]
I excluded the TSan & Level Zero failing tests in PR #1301, but the ASan failures seem genuine.

**Comment by pjaaskel** [comment]
@franz can you send the exclusions in a separate (single commit) PR so we can pull that in earlier? @jansol pls check the asans.

**Comment by pjaaskel** [comment]
```
26/27 Test #229: remote/test_command_buffer .................................***Failed  Required regular expression not found. Regex=[OK
] 14.05 sec
```

**Comment by pjaaskel** [comment]
@jansol  can you cherry-pick https://github.com/pocl/pocl/pull/1301/commits/ed882b691f53329932cec308d24e12ad0808c8f5 to this PR?

**Comment by pjaaskel** [comment]
```
 The following tests FAILED:
	222 - remote/test_buffer_migration (Failed)
	223 - remote/test_buffer_ping_pong (Failed)
	229 - remote/test_command_buffer (Failed)
Error: Process completed with exit code 8.
```

**Pull request merged**